feat: Finish integrations page

ivasilov opened the pull request:
This PR fixes several issues to finish the migration of the integration pages:
- Make the code block prettier.
- Add the `Admonition` component because it's used in some integration pages.
- Some pages might not have a `Learn` link, so don't show a section for if it's NULL.
- Move the OAuth and Integration (renamed to Marketplace) pages.
- Remove all integrations pages since they've been moved to the DB.

ivasilov committed:
Move the OAuth pages to the Platform guides.

ivasilov committed:
Fix the code block to be nicer.

ivasilov committed:
Add admonition component since it's required by some of the integration pages.

ivasilov committed:
Don't show the learn link if it's null.

ivasilov committed:
Move the generic integrations page to platform.

hieuwu commented on apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:
```diff
@@ -1060,6 +1061,30 @@ export const platform: NavMenuConstant = {
         { name: 'Production Checklist', url: '/guides/platform/going-into-prod' },
       ],
     },
+    {
```
Just wonder. In line 1032 you added a line break. If it is needed, please also add a line break here.
```suggestion
    
    {
```

hieuwu replied:
Great work! 👍

ivasilov committed:
Remove unneeded new line.

ivasilov committed:
Remove all specific integrations pages and their navigation.

kiwicopple commented:
<img width="356" alt="image" src="https://github.com/supabase/supabase/assets/10214025/2228c6bb-fb73-47a7-8e7d-bbeebff87eda">

looks like we removed the "Integrations" from the sidebar - perhaps we leave it there, but it can pop out to the partners site (in a new window)?

kiwicopple commented on apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:
```diff
@@ -1060,6 +1053,30 @@ export const platform: NavMenuConstant = {
         { name: 'Production Checklist', url: '/guides/platform/going-into-prod' },
       ],
     },
+    {
+      name: 'OAuth Apps (Beta)',
+      url: undefined,
+      items: [
+        {
+          name: 'Publish an OAuth App',
+          url: '/guides/platform/oauth-apps/publish-an-oauth-app',
+        },
+        {
+          name: 'Authorize an OAuth App',
+          url: '/guides/platform/oauth-apps/authorize-an-oauth-app',
+        },
+      ],
+    },
+    {
+      name: 'Integrations',
+      url: undefined,
+      items: [
+        {
+          name: 'Supabase Marketplace',
+          url: '/guides/platform/marketplace',
+        },
+      ],
+    },
```
```suggestion
    {
      name: 'Integrations',
      url: undefined,
      items: [
        {
          name: 'Integrations Marketplace',
          url: '/guides/platform/marketplace',
        },
        {
          name: 'Publish an OAuth App',
          url: '/guides/platform/oauth-apps/publish-an-oauth-app',
        },
        {
          name: 'Authorize an OAuth App',
          url: '/guides/platform/oauth-apps/authorize-an-oauth-app',
        },
      ],
    },
```

I'd just group all of these into one section - they all "fit" together pretty nicely

ivasilov replied:
Fixed.

gregnr reviewed (commented):
Nice one @ivasilov! Just a few things:
- Do we have a 1-to-1 mapping between old doc integration pages and new pages on partner site? We should add [redirects](https://github.com/supabase/supabase/blob/master/apps/www/lib/redirects.js) for each one (or maybe wildcard) so that any external links don't break
- We should do a quick search through the docs to see where we link to doc integrations and manually change to the new links (eg. [prisma reference in database docs](https://github.com/supabase/supabase/blob/bc0a41618e3dc930f23bd997407527d76720244a/apps/docs/pages/guides/database/connecting-to-postgres.mdx#L41))
- Did you end up getting CodeHike to render properly on the partner pages? We've recently been making a switch away from the `CodeBlock` component in favour of code hike. Might be worth referencing how we do something similar [here](https://github.com/supabase/supabase/blob/93db839f885d93b3cb55a19a1ac342f178bacba4/apps/docs/pages/guides/ai/python/%5Bslug%5D.tsx#L129-L142). Happy to help!

ivasilov committed:
Fix the getStaticPaths to only include experts or technologies.

ivasilov committed:
Move the Admonition component to the ui package.

ivasilov committed:
Use CodeHike for rendering HTML Code in the www integration pages.

ivasilov committed:
Reorder the navigation items to put the integrations and Oauth apps under the same category.

ivasilov committed:
Add redirects from the old integrations to the new integrations pages.

ivasilov committed:
Switch the obsoleted routes to the integrations to the new routes.

ivasilov commented on apps/www/components/Admonition.tsx:
```diff
@@ -0,0 +1,50 @@
+  label?: string
+}
+
+const Admonition = ({ type = 'note', label, children }: PropsWithChildren<AdmonitionProps>) => {
```
Great suggestion! I've moved the `Admonition` component to the `ui` package and updated all its imports.

ivasilov commented:
> Nice one @ivasilov! Just a few things:
> 
>     * Do we have a 1-to-1 mapping between old doc integration pages and new pages on partner site? We should add [redirects](https://github.com/supabase/supabase/blob/master/apps/www/lib/redirects.js) for each one (or maybe wildcard) so that any external links don't break
> 
>     * We should do a quick search through the docs to see where we link to doc integrations and manually change to the new links (eg. [prisma reference in database docs](https://github.com/supabase/supabase/blob/bc0a41618e3dc930f23bd997407527d76720244a/apps/docs/pages/guides/database/connecting-to-postgres.mdx#L41))
> 
>     * Did you end up getting CodeHike to render properly on the partner pages? We've recently been making a switch away from the `CodeBlock` component in favour of code hike. Might be worth referencing how we do something similar [here](https://github.com/supabase/supabase/blob/93db839f885d93b3cb55a19a1ac342f178bacba4/apps/docs/pages/guides/ai/python/%5Bslug%5D.tsx#L129-L142). Happy to help!

Hey @gregnr, I did all your suggestions! Just a warning, the PR has gotten bigger :). 

I added redirects in `redirects.js` but I'm not sure if they work. They can probably only be tested on production because of their unique URL setup, right?

I updated all links in the docs and blog posts I could find. Let me know if I missed anything. I also switched the `pre` element to using `CodeHike`.

Let me know if I missed anything.

gregnr reviewed (approved):
@ivasilov Nice work 👍 looking good.

You can test redirects on staging via the `www` site:
https://zone-www-dot-com-git-feat-finish-integrations-page-supabase.vercel.app/docs/guides/integrations/auth0

Looks like its working.

Heads up, might have an image issue on the auth0 page:
![image](https://github.com/supabase/supabase/assets/4133076/90cea958-b906-4bcf-87a1-78b049c87d5b)

ivasilov commented:
> @ivasilov Nice work 👍 looking good.
> 
> You can test redirects on staging via the `www` site: https://zone-www-dot-com-git-feat-finish-integrations-page-supabase.vercel.app/docs/guides/integrations/auth0
> 
> Looks like its working.
> 
> Heads up, might have an image issue on the auth0 page: ![image](https://user-images.githubusercontent.com/4133076/255012740-90cea958-b906-4bcf-87a1-78b049c87d5b.png)

The image issues are part of the dynamic content (it's loaded via DB/Storage). I'll go over each page and fix that once I have this PR deployed, no worries.
